invite/views.py

```diff
@@ -95,7 +95,7 @@ def amnesia(request):
         form = forms.IForgotForm(request.POST)
         if form.is_valid():
             # determine user from email address.
-            user = User.objects.get(email__iexact=form.cleaned_data['email'])
+            user = User.objects.filter(email__iexact=form.cleaned_data['email'])[:1].get()
```

Comment:
I believe we'd prefer to get a single object rather than a slice, which would give us a singleton list. Also, don't think there's any need for the `.get()` at the end. I think this is more along the lines of what we'd want:
```suggestion
            user = User.objects.filter(email__iexact=form.cleaned_data['email'])[0]
```

Reply:
also, maybe we should add a test to cover this scenario?

Reply:
Yea...makes sense. Also I'm thinking I should probably add try...except here.

Reply:
what do you expect to catch?

Reply:
yes! and probably return None

Reply:
Also check if the user is None and prompt that user doesn't exist

Reply:
Since it is a view, it might make sense to catch the exception and return a response that says the email doesn't exist.

Reply:
Maybe instead of using try/except, just do a check on the size of the returned list before popping off the first element? something like:

```user = User.objects.filter(email__iexact=form.cleaned_data['email'])
if len(user):
    user = user[0]
else:
    return some error message
```

Reply:
Also, we _could_ use `first` instead of `filter`, which will return either one object or None.

Reply:
ah, well in any case, it seems like `first` would simplify things a bit, even without doing any additional checking. won't have to manually grab the zeroth object